### ISSUE TITLE
Automatic gradient with PyTorch and GPyTorch

### ISSUE THREAD

**Issue by takafusui** [cornellius-gp/gpytorch#1044]
Hi,

I use `GPyTorch` in the Gaussian process dynamic programming and use `PyTorch` to produce Jacobian by using its `torch.autograd.grad`.

My question here is how can I trace the automatic gradient information in GPyTorch.

I have the following script:
```python
k_plusplus = likelihood(gp_kplus(torch.tensor([x[0]], dtype=torch.float64)[None, :])).mean
```
where `x` is a variable with which I take gradient, and `likelihood` and `gp_kplus` are the usual likelihood function and the Gaussian process model from the `GPyTorch` library.
`print(k_plusplus)` returns:
```python
tensor([0.0711], dtype=torch.float64, grad_fn=<ViewBackward>)
```
It seems that gradient information is stored in the tensor object. However when I check with `torch.autograd.gradcheck`, there is a difference between the numerical and the analytical values...

Instead of using the Gaussian process regression, when I simplify the above part without changing the rest of my code:
```python
k_plusplus = torch.tensor(0.1, dtype=torch.float64)
```
It passes `torch.autograd.gradcheck`.

Do you have an idea how can I fix it?

Any help would be highly appreciated.

Thank you in advance.

**Comment by takafusui** [comment]
Sorry, this is my bad.
In the above code snippet, `x` is already a tensor object and I overwrite it by `torch.tensor([x[0]], dtype=torch.float64)` where I lost the gradient information, I guess.
When I modify:
```python
k_plusplus = likelihood(gp_kplus(x[None, :][0])).mean
```
Everything works.
